lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1219,6 +1223,20 @@ ChangeResult TypeAnalyzer::visitAten_SoftmaxLikeOp(
   return incorporateKnowledge(op.getResult(), knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenScalarImplicitOp(AtenScalarImplicitOp op) {
+  auto knowledge =
+      ValueKnowledge::getScalarPessimisticValueState(op.getContext());
+  BaseTensorType type = op.a().getType().cast<BaseTensorType>();
+  if (type.getDtype().isa<mlir::FloatType>()) {
+    knowledge.scalarType = type.getDtype();
```

Comment:
scalarType should be torch type here.

Reply:
Can you also add unit tests here test/Dialect/Torch/refine-types-ops.mlir?

Reply:
Here also https://github.com/llvm/torch-mlir/blob/main/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L1068-L1077 the scalarType is a builtin type.

Reply:
ha, I see. yea, those needs to be fixed as well.

Reply:
FYI, the fix and `setScalarType` are added in #838. Whichever PR merged later will need to solve the conflicts.